client/cli/src/commands/mod.rs

```diff
@@ -204,9 +204,10 @@ macro_rules! substrate_cli_subcommands {
 				&self,
 				base_path: &::std::path::PathBuf,
 				cache_size: usize,
+				database: crate::arg_enums::Database,
 			) -> $crate::Result<::sc_service::config::DatabaseConfig> {
 				match self {
-					$($enum::$variant(cmd) => cmd.database_config(base_path, cache_size)),*
+					$($enum::$variant(cmd) => cmd.database_config(base_path, cache_size, database)),*
 				}
 			}
 
```

Comment:
```suggestion

			fn database(&self) -> $crate::Result<::std::option::Option<$crate::Database>> {
				match self {
					$($enum::$variant(cmd) => cmd.database()),*
				}
			}

```

Always call the enum's variant implementation. Otherwise you get the default implementation instead of the enum's variant one.